src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CatalogPriceRuleController.php

```diff
@@ -289,4 +325,68 @@ private function getFormBuilder(): FormBuilderInterface
     {
         return $this->get('prestashop.core.form.identifiable_object.builder.catalog_price_rule_form_builder');
     }
+
+    /**
+     * @param CatalogPriceRuleList $catalogPriceRuleList
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function formatCatalogPriceRule(CatalogPriceRuleList $catalogPriceRuleList): array
+    {
+        $list = [];
+        foreach ($catalogPriceRuleList->getCatalogPriceRules() as $catalogPriceRule) {
+            $list[] = [
+                'id' => $catalogPriceRule->getCatalogPriceRuleId(),
+                'shop' => $catalogPriceRule->getShopName() ?? $this->trans('All shops', 'Admin.Global'),
+                'currency' => $catalogPriceRule->getCurrencyName() ?? $this->trans('All currencies', 'Admin.Global'),
+                'country' => $catalogPriceRule->getCountryName() ?? $this->trans('All countries', 'Admin.Global'),
+                'group' => $catalogPriceRule->getGroupName() ?? $this->trans('All groups', 'Admin.Global'),
+                'name' => $catalogPriceRule->getCatalogPriceRuleName(),
+                'fromQuantity' => $catalogPriceRule->getFromQuantity(),
+                'impact' => $this->formatImpact(
+                    $catalogPriceRule->getReductionType(),
+                    $catalogPriceRule->getReduction(),
+                    $catalogPriceRule->getCurrencyIso() ?: $this->getContextCurrencyIso(),
+                    $catalogPriceRule->getTaxIncl()
+                ),
+                'startDate' => $catalogPriceRule->getDateStart()->format(DateTimeUtil::DEFAULT_DATETIME_FORMAT),
+                'endDate' => $catalogPriceRule->getDateEnd()->format(DateTimeUtil::DEFAULT_DATETIME_FORMAT),
+            ];
+        }
+
+        return $list;
+    }
+
+    /**
+     * @param string $reductionType
+     * @param DecimalNumber $reductionValue
+     * @param string $currencyIsoCode
+     *
+     * @return string
+     */
+    private function formatImpact(
+        string $reductionType,
+        DecimalNumber $reductionValue,
+        string $currencyIsoCode,
+        bool $taxIncl
+    ): string
+    {
+        $impact = '';
+        if ($reductionValue->equalsZero()) {
+            return self::UNSPECIFIED_VALUE_FORMAT;
+        }
+
+        $reductionValue = $reductionValue->toNegative();
+
+        $locale = $this->getContextLocale();
+        if ($reductionType === Reduction::TYPE_AMOUNT) {
+            $price = $locale->formatPrice((string) $reductionValue, $currencyIsoCode);
+            if ($taxIncl) {
+                return $this->trans('%price% (tax incl.)', 'Admin.Global', ['%price%' => $price]);
+            }
+            return $this->trans('%price% (tax excl.)', 'Admin.Global', ['%price%' => $price]);
+        }
```

Comment:
Not sure about what domain should be here.

Reply:
Hey. It appears in catalog price rule list in product page. There is a column "Discount" that shows you what discount this catalog price rule gives, in case when it's an amount discount it can be either Tax Included or Tax Excluded
![image](https://user-images.githubusercontent.com/47781757/204220637-081b5f1c-8cfb-476b-9180-c1fa7e26a82b.png)